wps-common/src/main/java/au/org/aodn/aws/util/S3Utils.java

```diff
@@ -54,20 +54,54 @@ public static S3Object getS3Object(String s3Bucket, String s3Key) throws IOExcep
         return s3Client.getObject(s3Bucket, s3Key);
     }
 
-    public static void uploadToS3(File file, String s3bucket, String jobFileKey)
+    public static void uploadToS3(File file, String s3bucket, String fileKey, String contentType)
         throws InterruptedException {
         TransferManager tx = TransferManagerBuilder.defaultTransferManager();
-        Upload myUpload = tx.upload(s3bucket, jobFileKey, file);
-        myUpload.waitForCompletion();
-        tx.shutdownNow();
+        FileInputStream fileStream = null;
+        try {
+            Upload myUpload;
+
+            LOGGER.info("Uploading file [" + file.getAbsolutePath() + "] to S3 Bucket [" + s3bucket + "], Key [" + fileKey + "]");
+            if(contentType != null) {
+                ObjectMetadata meta = new ObjectMetadata();
+                meta.setContentType(contentType);
+                LOGGER.info("Setting contentType [" + contentType + "]");
+                fileStream = new FileInputStream(file);
+                myUpload = tx.upload(s3bucket, fileKey, fileStream, meta);
```

Comment:
I think you need to set the content-length otherwise the sdk will try and read the contents of the stream into memory first  to be able to work it out before uploading which won't be good for large files.  Refer https://github.com/aws/aws-sdk-java/issues/474#issuecomment-126413580